browser/lib/util/http.js

```diff
@@ -3,6 +3,16 @@ var Http = (function() {
 
 	function Http() {}
 
+	function shouldFallback(err) {
+		var statusCode = err.statusCode;
+		/* 400 + no code = a generic xhr onerror. Browser doesn't give us enough
+		 * detail to know whether it's fallback-fixable, but it may be (eg if a
+		 * network issue), so try just in case */
+		return (statusCode === 408 && !err.code) ||
+			(statusCode === 400 && !err.code)      ||
```

Comment:
Heh. Lines up correctly if viewed in an editor with tabstop=2 :P

lining up things on different levels of indentation when indented with tabs and the tabwidth might change isn't possible in general, but I'll kludge it with spaces...